From: Christoph Hellwig <hch@lst.de>
Date: Fri, 17 May 2019 02:47:34 -0700
Subject: nvme: remove the ifdef around nvme_nvm_ioctl
Patch-mainline: v5.2-rc2
Git-commit: 3f98bcc58cd5f1e4668db289dcab771874cc0920
References: git-fixes

We already have a proper stub if lightnvm is not enabled, so don't bother
with the ifdef.

Signed-off-by: Christoph Hellwig <hch@lst.de>
Reviewed-by: Keith Busch <keith.busch@intel.com>
Reviewed-by: Chaitanya Kulkarni <chaitanya.kulkarni@wdc.com>
Acked-by: Daniel Wagner <dwagner@suse.de>
---
 drivers/nvme/host/core.c |    2 --
 1 file changed, 2 deletions(-)

--- a/drivers/nvme/host/core.c
+++ b/drivers/nvme/host/core.c
@@ -1332,10 +1332,8 @@ static int nvme_ns_ioctl(struct nvme_ns
 	case NVME_IOCTL_SUBMIT_IO:
 		return nvme_submit_io(ns, (void __user *)arg);
 	default:
-#ifdef CONFIG_NVM
 		if (ns->ndev)
 			return nvme_nvm_ioctl(ns, cmd, arg);
-#endif
 		if (is_sed_ioctl(cmd))
 			return sed_ioctl(ns->ctrl->opal_dev, cmd,
 					 (void __user *) arg);
